packages/build-tools/src/utils/expoUpdates.ts

```diff
@@ -88,14 +94,36 @@ export async function configureExpoUpdatesIfInstalledAsync(
   }
 
   const appConfigRuntimeVersion = ctx.job.version?.runtimeVersion ?? resolvedRuntimeVersion;
+
   if (ctx.metadata?.runtimeVersion && ctx.metadata.runtimeVersion !== appConfigRuntimeVersion) {
     ctx.markBuildPhaseHasWarnings();
-    ctx.logger.warn(
-      `Runtime version from the app config evaluated on your local machine (${ctx.metadata.runtimeVersion}) does not match the one resolved here (${appConfigRuntimeVersion}).`
-    );
-    ctx.logger.warn(
-      "If you're using conditional app configs, e.g. depending on an environment variable, make sure to set the variable in eas.json or configure it with EAS Secret."
-    );
+    ctx.logger.warn('Runtime version mismatch');
+    ctx.logger.warn(`Runtime version on your local machine: ${ctx.metadata.runtimeVersion}`);
+    ctx.logger.warn(`Runtime version calculated on EAS: ${appConfigRuntimeVersion}`);
+
+    if (ctx.metadata?.fingerprintSource && resolvedFingerprintSources && resolvedRuntimeVersion) {
+      try {
+        const fingerprintSource = ctx.metadata?.fingerprintSource;
+
+        if (fingerprintSource.type === FingerprintSourceType.URL) {
+          const result = await fetch(fingerprintSource.url);
+          const localFingerprint = await result.json();
+          const easFingerprint = {
+            hash: resolvedRuntimeVersion,
+            sources: resolvedFingerprintSources as FingerprintSource[],
+          };
+          const changes = diffFingerprints(localFingerprint, easFingerprint);
+          if (changes.length) {
+            ctx.logger.warn('Difference between local and EAS fingerprints:');
+            ctx.logger.warn(stringifyFingerprintDiff(changes));
+          }
+        } else if (fingerprintSource.type === FingerprintSourceType.PATH) {
+          // TODO(Kadi): handle local fingerprint path files
```

Comment:
I think that the only thing to do here when it comes to loading files will be
```suggestion
          const localFingerprint = await fs.readJson(fingerprintSource.path);
          // ... additional parsing and diffing as above
```